packages/frontend-shared/src/utils/getUrlWithParams.ts

```diff
@@ -7,14 +7,29 @@ export type LinkWithParams = {
 
 export const getUrlWithParams = (link: LinkWithParams) => {
   let result = link.url
-  const hasUtmParams = Object.keys(link.params).some((param) => param.startsWith('utm_'))
 
-  if (hasUtmParams) {
-    link.params.utm_source = getUtmSource()
-  }
+  if (Object.keys(link.params).length > 0) {
```

Comment:
Nit: You can store `Object.keys(link.params)` so you don't have to call it twice.